elastic/kibana
# [Controls] Add Expensive Queries Fallback

## Summary
Fixes https://github.com/elastic/kibana/issues/155078.

In main right now if you set up a user with `read` only access to an index controls will default to using the non-expensive queries, and some features will be missing. This is because without the `monitor` privilege, it is impossible to check whether expensive queries are on or off.

Because the default setting for `allow_expensive_queries` is `true`, we should default to `true` for cases where a security exception prevents us from checking the value of the setting.

## How to test
- Install the web logs sample data
- Add a role that has Dashboard read permission and `read` permission for the logs web traffic index.
- Log in as the user and load up the dashboard.
- You should not see the warning and the control should work as expected

## Open question
Should we change this default behaviour? 
- With this change, in the unlikely scenario that a cluster has `allow_expensive_queries` off, and the user cannot check the value of the setting due to a missing permission, the Controls will throw errors. This is similar to the way the table list view and solutions work - they don't even check for the value of the setting. 
- Without this change, we are requiring analysts who use the controls to have the `monitor` permission.
